src/core/server/services/slack/publisher.ts

```diff
@@ -0,0 +1,190 @@
+import { CommentLeftModerationQueueInput } from "coral-server/graph/tenant/resolvers/Subscription/commentLeftModerationQueue";
+import { CommentReleasedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentReleased";
+import { CommentReplyCreatedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentReplyCreated";
+import { CommentStatusUpdatedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentStatusUpdated";
+import { SUBSCRIPTION_CHANNELS } from "coral-server/graph/tenant/resolvers/Subscription/types";
+import logger from "coral-server/logger";
+
+import { GQLMODERATION_QUEUE } from "coral-server/graph/tenant/schema/__generated__/types";
+
+import { Tenant } from "coral-server/models/tenant";
+import { Db } from "mongodb";
+import SlackContext from "./context";
+
+type Payload =
+  | CommentEnteredModerationQueueInput
+  | CommentLeftModerationQueueInput
+  | CommentStatusUpdatedInput
+  | CommentReplyCreatedInput
+  | CommentCreatedInput
+  | CommentFeaturedInput
+  | CommentReleasedInput;
+
+function enteredModeration(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    (p.queue === GQLMODERATION_QUEUE.REPORTED ||
+      p.queue === GQLMODERATION_QUEUE.PENDING)
+  );
+}
+
+function isFeatured(channel: SUBSCRIPTION_CHANNELS) {
+  return channel === SUBSCRIPTION_CHANNELS.COMMENT_FEATURED;
+}
+
+function isReported(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    p.queue === GQLMODERATION_QUEUE.REPORTED
+  );
+}
+
+function isPending(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    p.queue === GQLMODERATION_QUEUE.PENDING
+  );
+}
+
+function createModerationLink(rootURL: string, commentID: string) {
+  return `${rootURL}/admin/moderate/comment/${commentID}`;
+}
+
+async function postCommentToSlack(
+  ctx: SlackContext,
+  orgURL: string,
+  commentID: string,
+  webhookURL: string
+) {
+  const comment = await ctx.comments.load(commentID);
+  if (comment === null) {
+    return;
+  }
+  const story = await ctx.stories.load(comment.storyID);
+  if (story === null) {
+    return;
+  }
+  const author = await ctx.users.load(comment.authorID);
+  if (author === null) {
+    return;
+  }
+
+  const storyTitle = story.metadata ? story.metadata.title : "";
+  const commentBody =
+    comment.revisions.length > 0
+      ? comment.revisions[comment.revisions.length - 1].body
+      : "";
+  const body = commentBody.replace(new RegExp("<br>", "g"), "\n");
+  const moderateLink = createModerationLink(orgURL, comment.id);
+
+  const data = {
```

Comment:
When building this, do we want to have some kind of templating?
Currently, this will only support English and I could easily see a way to template this with string interpolation and substitution.

Reply:
I think in a future iteration we can tie this into i18n, but for now this should meet our requirements.